src/main/java/com/epam/training/tasks/third/entities/CubeObservable.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.training.tasks.third.entities;
+
+import com.epam.training.tasks.third.logic.CubeIdGenerator;
+import org.apache.log4j.Logger;
+
+
+import java.util.List;
+import java.util.Objects;
+
+
+public class CubeObservable extends Cube implements Observable {
+
+    private static final Logger LOGGER = Logger.getLogger(CubeObservable.class);
+
+    private List<CubeObserver> observers;
+    private final CubeIdGenerator idGenerator = new CubeIdGenerator();
+    private int id;
+
+
+    public CubeObservable(double x, double y, double z, double sideLength) {
+        super(x, y, z, sideLength);
+        setId(this);
+    }
+
+    public CubeObservable(int id, double x, double y, double z, double sideLength) {
+        super(x, y, z, sideLength);
+        this.id = id;
+    }
+
+    public CubeObservable(Point point, double sideLength) {
+        super(point, sideLength);
+        setId(this);
+    }
+
+    public CubeObservable(double[] data) {
+        super(data);
+        setId(this);
+    }
+
+    private void setId (CubeObservable cube) {
+
+        int id = idGenerator.getNewId();
+        cube.id = id;
+    }
+
+
+    public int getId() {
+        return this.id;
+    }
+
+    public void setNewParams(int id, double x, double y, double z, double sideLength) {
```

Comment:
don't invent something, use standard set methods (setX, setY, setSideLength) and override them in this class by calling super.set and then notify